### PR TITLE
feat(ec2): ESP and AH IPsec protocols for Security Groups

### LINKED ISSUE #13403: (aws-ec2): cannot create ingress rule for protocols ESP or AH
We need ingress and egress rules for protocol 50 (ESP) and 51 (AH), `ec2.Port` has an extremely limited set of protocol values it allows.

### Reproduction Steps

```python
#!/usr/bin/env python3

from aws_cdk import core, aws_ec2 as ec2


app = core.App()
stack = core.Stack()
vpc = ec2.Vpc(stack, "VPC", cidr="10.0.0.0/16")
vpn = ec2.SecurityGroup(
    stack,
    "VPNSecurityGroup",
    vpc=vpc,
    description="External VPN",
)
vpn.add_ingress_rule(peer=ec2.Peer.ipv4("0.0.0.0/0"), connection=ec2.Port(protocol="50", string_representation="ESP"), description="protocol 50, ESP")
vpn.add_ingress_rule(peer=ec2.Peer.ipv4("0.0.0.0/0"), connection=ec2.Port(protocol="51", string_representation="AH"), description="protocol 51, AH")

app.synth()
```

### What did you expect to happen?
Should be able to add legal CFN rules.

### What actually happened?

```
$ cdk synth
jsii.errors.JavaScriptError:
  Error: Expected enum value, got "50"
      at Object.deserialize (/private/var/folders/v1/y34vbhp158x18m1dqjzb57mw0001yl/T/tmp3q3mzj3r/lib/program.js:3508:31)
      at Kernel._toSandbox (/private/var/folders/v1/y34vbhp158x18m1dqjzb57mw0001yl/T/tmp3q3mzj3r/lib/program.js:3071:69)
      at /private/var/folders/v1/y34vbhp158x18m1dqjzb57mw0001yl/T/tmp3q3mzj3r/lib/program.js:3636:37
      at mapValues (/private/var/folders/v1/y34vbhp158x18m1dqjzb57mw0001yl/T/tmp3q3mzj3r/lib/program.js:3907:35)
      at Object.deserialize (/private/var/folders/v1/y34vbhp158x18m1dqjzb57mw0001yl/T/tmp3q3mzj3r/lib/program.js:3632:28)
      at Kernel._toSandbox (/private/var/folders/v1/y34vbhp158x18m1dqjzb57mw0001yl/T/tmp3q3mzj3r/lib/program.js:3071:69)
      at /private/var/folders/v1/y34vbhp158x18m1dqjzb57mw0001yl/T/tmp3q3mzj3r/lib/program.js:3119:42
      at Array.map (<anonymous>)
      at Kernel._boxUnboxParameters (/private/var/folders/v1/y34vbhp158x18m1dqjzb57mw0001yl/T/tmp3q3mzj3r/lib/program.js:3119:27)
      at Kernel._toSandboxValues (/private/var/folders/v1/y34vbhp158x18m1dqjzb57mw0001yl/T/tmp3q3mzj3r/lib/program.js:3105:29)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "app.py", line 15, in <module>
    vpn.add_ingress_rule(peer=ec2.Peer.ipv4("0.0.0.0/0"), connection=ec2.Port(protocol="50", string_representation="ESP"), description="protocol 50, ESP")
  File "/Users/aw/Projects/cureatr/dev/cureatr/ops/ec2/protocol/.venv/lib/python3.7/site-packages/jsii/_runtime.py", line 83, in __call__
    inst = super().__call__(*args, **kwargs)
  File "/Users/aw/Projects/cureatr/dev/cureatr/ops/ec2/protocol/.venv/lib/python3.7/site-packages/aws_cdk/aws_ec2/__init__.py", line 37646, in __init__
    jsii.create(Port, self, [props])
  File "/Users/aw/Projects/cureatr/dev/cureatr/ops/ec2/protocol/.venv/lib/python3.7/site-packages/jsii/_kernel/__init__.py", line 281, in create
    for iface in getattr(klass, "__jsii_ifaces__", [])
  File "/Users/aw/Projects/cureatr/dev/cureatr/ops/ec2/protocol/.venv/lib/python3.7/site-packages/jsii/_kernel/providers/process.py", line 344, in create
    return self._process.send(request, CreateResponse)
  File "/Users/aw/Projects/cureatr/dev/cureatr/ops/ec2/protocol/.venv/lib/python3.7/site-packages/jsii/_kernel/providers/process.py", line 326, in send
    raise JSIIError(resp.error) from JavaScriptError(resp.stack)
jsii.errors.JSIIError: Expected enum value, got "50"
Subprocess exited with error 1
```

### Environment

  - **CDK CLI Version  :** 1.91.0 (build 0f728ce)
  - **Framework Version:** 1.91.0
  - **Node.js Version:** v14.15.0
  - **OS               :** macOS 10.15.7
  - **Language (Version):** Python 3.7.9

### Other

Make `ec2.Port` `protocol` a string instead of a limited enum

--- 

This is :bug: Bug Report

### PR DESCRIPTION
**Pull request by hollanddd**
First contribution. I've gone through the checklist and think i've managed to hit all the requirements. I'd like to contribute more and I'm taking it slow so let me know how I can improve my PRs

Closes #13403 

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by rix0rrr — APPROVED** [review]
This is perfect, thanks!

File: packages/@aws-cdk/aws-ec2/lib/port.ts [hunk 1]
```diff
@@ -9,6 +9,8 @@ export enum Protocol {
   UDP = 'udp',
   ICMP = 'icmp',
   ICMPV6 = '58',
+  ESP = 'esp',
+  AH = 'ah',
```
Comment (rectalogic) [inline packages/@aws-cdk/aws-ec2/lib/port.ts:13]: This needs to be `ESP = ‘50’` and  `AH = ‘51’`

File: packages/@aws-cdk/aws-ec2/lib/port.ts [hunk 2]
```diff
@@ -171,6 +173,30 @@ export class Port {
...
+  }
+
+  /**
+   * A single AH port
+   */
+  public static ah(): Port {
+    return new Port({
+      protocol: Protocol.AH,
+      fromPort: 51,
+      toPort: 51,
```
Comment (rectalogic) [inline packages/@aws-cdk/aws-ec2/lib/port.ts:195]: fromPort and toPort both need to be unset for both protocols

File: packages/@aws-cdk/aws-ec2/lib/port.ts (hunk 1 above)
Comment (rectalogic) [reply packages/@aws-cdk/aws-ec2/lib/port.ts:13]: Also there are many other protocols that need to be added to this enum
https://en.wikipedia.org/wiki/List_of_IP_protocol_numbers

**Pull request merged**